app/services/exports/audience_export_service.rb

```diff
@@ -3,17 +3,42 @@
 require "csv"
 
 class Exports::AudienceExportService
-  def initialize(user)
+  FIELDS = ["Subscriber Email", "Subscribed Time"].freeze
+
+  def initialize(user, options = {})
     @user = user
+    @options = options.with_indifferent_access
+    timestamp = Time.current.to_fs(:db).gsub(/ |:/, "-")
+    @filename = "Subscribers-#{@user.username}_#{timestamp}.csv"
   end
 
+  attr_reader :filename, :tempfile
+
   def perform
-    CSV.generate do |csv|
-      csv << ["Follower Email", "Followed Time"]
+    @tempfile = Tempfile.new(["Subscribers", ".csv"], "tmp", encoding: "UTF-8")
+
+    CSV.open(@tempfile, "wb") do |csv|
+      csv << FIELDS
+
+      query = @user.audience_members.select(:id, :email, :min_created_at)
+
+      conditions = []
+      conditions << "follower = true" if @options[:followers]
+      conditions << "customer = true" if @options[:customers]
+      conditions << "affiliate = true" if @options[:affiliates]
 
-      @user.followers.active.find_each do |follower|
-        csv << [follower.email, follower.created_at]
+      query = query.where(conditions.join(" OR "))
+
+      query.order(:min_created_at).each do |member|
```

Comment:
[`find_each`](https://edgeapi.rubyonrails.org/classes/ActiveRecord/Batches.html#method-i-find_each) instead of `each` to avoid loading everything into memory at once.